apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/DatasetExtensions.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright 2020 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Apis.Bigquery.v2.Data;
+using System;
+
+namespace Google.Cloud.BigQuery.V2
+{
+    /// <summary>
+    /// Extension methods for making it easier to work with <see cref="Dataset"/>
+    /// </summary>
+    public static class DatasetExtensions
+    {
+        /// <summary>
+        /// Gets the default expiration period for tables created in this dataset.
+        /// Tables will (by default) be automatically deleted this long after they are created,
+        /// unless otherwise specified, such as by <see cref="CreateTableOptions.Expiration"/>.
+        /// </summary>
+        /// <param name="dataset">The dataset for which to set the expiration for.
+        /// Must not be null.</param>
+        public static TimeSpan? GetDefaultTableExpiration(this Dataset dataset) =>
+            GaxPreconditions.CheckNotNull(dataset, nameof(dataset)).DefaultTableExpirationMs.HasValue ?
+            TimeSpan.FromMilliseconds(dataset.DefaultTableExpirationMs.Value) : (TimeSpan?)null;
+
+        /// <summary>
+        /// Sets the default expiration period for tables created in this dataset.
+        /// Tables will (by default) be automatically deleted this long after they are created,
+        /// unless otherwise specified, such as by <see cref="CreateTableOptions.Expiration"/>.
+        /// </summary>
+        /// <param name="dataset">The dataset for which to set the expiration for.
+        /// Must not be null.</param>
+        /// <param name="expiration">The new expiration to be set.</param>
+        public static void SetDefaultTableExpiration(this Dataset dataset, TimeSpan? expiration) =>
```

Comment:
This could potentially return `dataset` for easier method chaining. (Ditto SetDefaultPartitionExpiration).